### PR TITLE
Post new message

### PR DESCRIPTION
**Pull request by jans510**

### CODE REVIEW

File: src/app/features/chat/page/chat-page/chat-page.component.html [hunk 1]
```diff
@@ -7,25 +7,18 @@ <h1>Deine Chats</h1>
     <mat-grid-tile>
       <div>
         <div *ngFor="let chat of chatService.chats$ | async">
```
Comment (halbekanne) [inline src/app/features/chat/page/chat-page/chat-page.component.html:9]: I see you're using 3 async pipes in your template. You can simplify this in a nice way by using the `combineLatest` operator from RxJs: https://dev.to/danywalls/how-to-simplify-multiple-async-pipes-39di

Another option I saw is to combine the async pipes into one statement, like so:
```
<ng-container *ngIf="
  {
    obs1: obs1$ | async,
    obs2: obs2$ | async,
    obs3: obs3$ | async
  } as observables">
    <!--    already expanded observable!-->
    {{observables.obs1}}
</ng-container>
```

File: src/app/features/chat/components/chat-messages/chat-messages.component.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  @Input() selectedChat?: Chat;
+  @Input() messagesByChat?: MessagesByChatMap;
+
+  constructor(
+    private fb: NonNullableFormBuilder,
+    public chatService: ChatService
+  ) {}
+
+  ngOnInit(): void {
+    this.addMessageForm = this.fb.group({
```
Comment (halbekanne) [inline src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: A collegue of mine told me you can better do assignments like that at the point of variable declaration, not in the initialization method. At the point when you define `addMessageForm` in line 16, you already have access to all classes injected in the constructor. So you could just directly define it in line 16. Another bonus is you don't have to use the any type.

**Review by halbekanne — CHANGES_REQUESTED** [review]

File: src/app/features/chat/common/services/chat.service.ts [hunk 3]
```diff
@@ -35,11 +38,29 @@ export class ChatService {
...
       if (response.isLoggedIn) {
-        this.fetchChatsAndDancers();
+        this.pollForChats();
+      } else {
+        this.stopPollingForChats();
       }
     });
   }
 
+  createAndFetchMessages$(text: string): Observable<APIResponse<void>> {
```
[the remark this answers is not in the document]
Comment (jans510) [reply src/app/features/chat/common/services/chat.service.ts:48]: hmm, this then needs changes also in chat dancer and dancer as we do not return any payload or location header right now. I could add it but I would do it in a separate PR.

File: src/app/features/chat/components/chat-messages/chat-messages.component.ts (hunk 2 above)
Comment (jans510) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: it looks like the NonNullableFormBuilder is not defined at this stage. I get an undefined exception if I do it as you suggested

Comment (halbekanne) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: Have a look at src/app/features/profile/edit/pages/init-user-name/init-user-name.component.ts it should be possible to do it like that. I don't understand why it doesn't work in your case.

Comment (jans510) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: for some reason I get an undefined error in every component that is using the approach you mentioned. I get it also in the init-username component

Comment (halbekanne) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: You're using IntelliJ too right? Does the problem only show up in the editor or during compilation too, e.g. does the error prevent you from running `ng serve` and rendering the site locally? If it only shows up in the editor and nowhere else, this must be a configuration issue then. Maybe you can google the exact error message or reimport the project somehow.

Comment (jans510) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: no, I only get it running show dancer and opening it in the browser. It also happens for me on chrome and safari using the test environment. Have you maybe updated angular recently? Can you try this link and check if it works for you or if you also get an error?
https://test.dancier.net/registration/verify-account/c6272287-7a4b-4ef9-92b2-c7dd967c1171
The error is: `main.1e765d5193d6558a.js:1 ERROR Error: Uncaught (in promise): TypeError: Cannot read properties of undefined (reading 'group'`

Comment (jans510) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: could be related: https://stackoverflow.com/questions/74774468/angular-15-formsbuilder-typeerror-cannot-read-properties-of-undefined-read

Comment (halbekanne) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: Ok I can see the problem now... no idea what happened there, normally it must be initialized when used inside a constructor. I'll try to update to v15 just to see if this fixes something.

Comment (halbekanne) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: The update process for Angular 15 talked about the tsconfig parameter `useDefineForClassFields` that the linked StackOverflow also mentioned.

```
❯ Update TypeScript compiler `target` and set `useDefineForClassFields`.
  These changes are for IDE purposes as TypeScript compiler options `target` and `useDefineForClassFields` are set to `ES2022` and `false` respectively by the Angular CLI.
```

Updating to the latest Angular Version solved the issue. However, updating has unintended consequences as some dependencies are deprecated as are some Material components. Updating breaks the UI of the website...

I think for this PR leave it as is and I'm going to see if I can do the Angular update in a separate PR and fix everything.

Comment (jans510) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: the problem is that everywhere where we use this approach we have broken logic. I think the root cause was me adding es2022 compileroptions in the tsconfig in december: https://angular.schule/blog/2022-11-use-define-for-class-fields
Unfortunately setting useDefineForClassFields to false breaks other things ...

Comment (halbekanne) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: Upgrading to v15 wasn't it. `"useDefineForClassFields": false,` in tsconfig.json fixes this issue in my testing... however I don't know if this is the right long-term solution.

Comment (halbekanne) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: > the problem is that everywhere where we use this approach we have broken logic. I think the root cause was me adding es2022 compileroptions in the tsconfig in december: https://angular.schule/blog/2022-11-use-define-for-class-fields Unfortunately setting useDefineForClassFields to false breaks other things ...

Oh... well that's unfortunate. Maybe we should change the target back then.

Comment (halbekanne) [reply src/app/features/chat/components/chat-messages/chat-messages.component.ts:28]: Call me and we can talk about it to find the best solution for now.

**Review by halbekanne — APPROVED** [review]

**Pull request merged**